### ISSUE TITLE
Create snapshot on source for Data SYnc

### ISSUE THREAD

**Issue by atulwy** [webyog/sqlyog-community#2055]
Original [issue 2056](https://code.google.com/p/sqlyog/issues/detail?id=2056) created by webyog on 2014-10-06T07:39:26.000Z:

In Data Sync we SELECT ... LOCK IN SHARE MODE on source in order to ensure consistency. This however only ensures consistency if enforeced using Foreign Keys in the database. Applications may have requirements for consistency that are not enforced in the database, but in application code. Data Sync cannot know the requirements of applications and may break such requirements.

With InnoDB we can create a snapshot of all tables by executing &quot;SET SESSION TRANSACTION ISOLATION LEVEL REPEATABLE READ; START TRANSACTION;&quot; and I think we should have this as an option.

Basically I think this would be a logical enhancement to utilize what features InnoDB offers.

(this is a 'spin-off' from discussion in ticket /48524522/68190/)
